R/generateHypotheses.R

```diff
@@ -1,192 +1,305 @@
+#' ##Inspect the result
+#' binaryInteractions
+#' 
+#' 
+generateBinaryNetwork <- function(complex_hypotheses){
+  setkey(complex_hypotheses, "complex_id")
+  complexes <- unique(complex_hypotheses$complex_id)
+  ## Get every combination of binary interactions within a complex
+  binary_interactions <- do.call("rbind", lapply(complexes, function(x){
+    t(combn(complex_hypotheses[complex_id == x, protein_id], m = 2))
+  }))
+  ## Keep only unique interactions
+  binary_interactions <- t(apply(binary_interactions,1, sort))
   binary_interactions <- unique(binary_interactions)
-  binary_interactions[]
+  
+  binary_interactions <- as.data.table(binary_interactions)
+  names(binary_interactions) <- c("a", "b")
+  setkey(binary_interactions,NULL)
+  
+  return(binary_interactions)
 }
 
-#' appendBinaryInteractions
-#' @description Append binary interaction table with additional complexes
-#' @import data.table
-#' @param complex character string with complex_id
-#' @param hypothesis data.table in the format of complex hypotheses
-#' @param data.table with binary interactions between "a" and "b"
-#' @return data.table with binary interactions between "a" and "b" appended by current complex interactions
-appendBinaryInteractions <- function(complex,hypothesis,binary_interactions){
-  complex_proteins <- unique(hypothesis$protein_id[which(hypothesis$complex_id == complex)]) # get all proteins within complex
-  binaries <- as.list(data.table(combn(complex_proteins,2))) # create all binary combinations of the proteins in complex
-  binaries_sort <- lapply(binaries,FUN=sort) # sort alphanumerically to make it possible to remove duplicates at the end
-  binaries_sort_table <- data.table(t(as.data.table(binaries_sort)))
-  colnames(binaries_sort_table) <- c("a","b")
-  binary_interactions <- rbind(binary_interactions,binaries_sort_table)
-  binary_interactions
-}
 
-#' calculatePathlength
+#' Calculate pathlength from binary interactions
 #' @description Calculate shortest paths between all members of a binary interaction network.
 #' This function creates a table that contains all possible combinations of 2 from
-#' all input proteins (binomial coeffitient (length(proteins) over 2)).
+#' all input proteins (binomial coefficient (length(proteins) over 2)).
 #' @import data.table
-#' @import igraph
+#' @importFrom igraph graph_from_data_frame
+#' @importFrom igraph distances
 #' @param binaryInteractionTable data.table with binary interactions between "a" and "b" as returned from generateBinaryNetwork()
-#' @return data.table with three columns (x,y,dist) containing all possible combinations
-#' of proteins and their shortest distance as determined by the input binary interaction table
+#' @return data.table with three columns containing all possible combinations of proteins:
+#' \itemize{
+#'   \item x,y: Protein ids of the input proteins (all possible combinations)
+#'   \item dist: distance as determined by the input binary interaction table
```

Comment:
maybe add the definition of the distance: shortest path length between the two proteins